python-checks/src/main/java/org/sonar/python/checks/BooleanExpressionInExceptCheck.java

```diff
@@ -37,10 +57,62 @@ public void initialize(Context context) {
 
   private static void checkExceptClause(SubscriptionContext ctx) {
     ExceptClause except = (ExceptClause) ctx.syntaxNode();
-    Expression exception = Expressions.removeParentheses(except.exception());
-    if (exception != null && exception.is(Kind.OR, Kind.AND)) {
-      ctx.addIssue(exception, "Rewrite this \"except\" expression as a tuple of exception classes.");
+    Optional.of(except)
+      .map(ExceptClause::exception)
+      .map(Expressions::removeParentheses)
+      .filter(exception -> exception.is(Kind.OR, Kind.AND))
+      .ifPresent(exception -> {
+        var issue = (IssueWithQuickFix) ctx.addIssue(exception, MESSAGE);
+        addQuickFix(issue, exception);
+      });
+  }
+
+  private static List<String> collectNames(Tree expression) {
+    if (expression.is(Kind.OR, Kind.AND)) {
+      var binaryExpression = (BinaryExpression) expression;
+      var leftExceptions = collectNames(binaryExpression.leftOperand());
+      var rightExceptions = collectNames(binaryExpression.rightOperand());
+      var result = new ArrayList<String>();
+      result.addAll(leftExceptions);
+      result.addAll(rightExceptions);
+      return result;
+    } else if (expression.is(Kind.PARENTHESIZED)) {
+      var parenthesizedExpression = (ParenthesizedExpression) expression;
```

Comment:
You can actually use `Expressions#removeParentheses` here as well (result will be exactly the same).
You could probably do that as soon as you receive the `expression` in the method so you don't have to deal with it in the `if / else if` chain.

Reply:
done